lib/iris/_concatenate.py

```diff
@@ -377,33 +462,34 @@ def _coordinate_differences(self, other, attr):
             between `self` and `other`.
 
         Returns:
-            Tuple of a descriptive error message and the names of coordinates
-            that differ between `self` and `other`.
+            List of descriptive error messages of the differences between
+            `self` and `other`.
 
         """
         # Set up {name: coord_metadata} dictionaries.
         self_dict = {x.name(): x for x in getattr(self, attr)}
         other_dict = {x.name(): x for x in getattr(other, attr)}
         if len(self_dict) == 0:
-            self_dict = {'< None >': None}
+            self_dict = {'None': None}
         if len(other_dict) == 0:
-            other_dict = {'< None >': None}
+            other_dict = {'None': None}
         self_names = sorted(self_dict.keys())
         other_names = sorted(other_dict.keys())
 
         # Compare coord metadata.
         if len(self_names) != len(other_names) or self_names != other_names:
```

Comment:
This test is unnecessarily complex : 
I think just `self_names != other_names` does it all ?